kedro-datasets/kedro_datasets/databricks/managed_table_dataset.py

```diff
@@ -0,0 +1,463 @@
+                    f"has columns: {base_columns}, whereas "
+                    f"dataframe has columns {update_columns}"
+                )
+
+            where_expr = ""
+            if isinstance(self._table.primary_key, str):
+                where_expr = (
+                    f"base.{self._table.primary_key}=update.{self._table.primary_key}"
+                )
+            elif isinstance(self._table.primary_key, list):
+                where_expr = " AND ".join(
+                    f"base.{col}=update.{col}" for col in self._table.primary_key
+                )
+
+            update_data.createOrReplaceTempView("update")
+            self._get_spark().conf.set(
+                "fullTableAddress", self._table.full_table_location()
+            )
+            self._get_spark().conf.set("whereExpr", where_expr)
+            upsert_sql = """MERGE INTO ${fullTableAddress} base USING update ON ${whereExpr}
+                WHEN MATCHED THEN UPDATE SET * WHEN NOT MATCHED THEN INSERT *"""
+            self._get_spark().sql(upsert_sql)
+        else:
+            self._save_append(update_data)
+
+    def _save(self, data: Union[DataFrame, pd.DataFrame]) -> None:
+        """Saves the data based on the write_mode and dataframe_type in the init.
+        If write_mode is pandas, Spark dataframe is created first.
+        If schema is provided, data is matched to schema before saving
+        (columns will be sorted and truncated).
+
+        Args:
+            data (Any): Spark or pandas dataframe to save to the table location
+        """
+        # filter columns specified in schema and match their ordering
+        if self._table.schema():
+            cols = self._table.schema().fieldNames()
+            if self._table.dataframe_type == "pandas":
+                data = self._get_spark().createDataFrame(
+                    data.loc[:, cols], schema=self._table.schema()
+                )
+            else:
+                data = data.select(*cols)
+        else:
+            if self._table.dataframe_type == "pandas":
+                data = self._get_spark().createDataFrame(data)
+        if self._table.write_mode == "overwrite":
+            self._save_overwrite(data)
+        elif self._table.write_mode == "upsert":
+            self._save_upsert(data)
+        elif self._table.write_mode == "append":
+            self._save_append(data)
+
+    def _describe(self) -> Dict[str, str]:
+        """Returns a description of the instance of ManagedTableDataSet
+
+        Returns:
+            Dict[str, str]: Dict with the details of the dataset
+        """
+        return {
+            "catalog": self._table.catalog,
+            "database": self._table.database,
+            "table": self._table.table,
+            "write_mode": self._table.write_mode,
+            "dataframe_type": self._table.dataframe_type,
+            "primary_key": self._table.primary_key,
+            "version": self._version,
```

Comment:
```suggestion
            "version": str(self._version),
```

For consistency with the generic `Dict[str, str]`.